### ISSUE TITLE
Use of deprecated BinarySerializers trips built fault with dotnet 5.0.0

### ISSUE THREAD

**Issue by RobertHenry6bev** [dotnet/performance#1505]
There are ~6 places in the dotnet benchmarks that use the soon-to-be deprecated binary serializer.  Here's one:

https://github.com/dotnet/performance/blob/28c51901800c47c8715b3d8df5305206aba1a631/src/benchmarks/micro/Serializers/Binary_FromStream.cs#L62

That triggers an error when built with dotnet 5.0.0

See https://github.com/GrabYourPitchforks/docs/blob/bf_obsoletion_docs/docs/standard/serialization/resolving-binaryformatter-obsoletion-errors.md

**Comment by danmosemsft** [comment]
Did https://github.com/dotnet/performance/pull/1403 not do the trick?

https://github.com/dotnet/performance/issues/1402

**Comment by RobertHenry6bev** [comment]
I'm working against head of master, and I believe the change #1403 is applied therein.  I still get the problem.

I'm trying to run the aspnet benchmark ensemble using the latest 5.0.0.preview-8 on an ARM64 (aarch64) linux host.  The test jig runs client code inside of a docker image.

I'm encountering situations where the various dotnet/aspnet benchmarks implicitly or explicitly refer to amd64(eg x86_64) binaries, rather than ARM64 binaries.  Or else reference packages on the web that don't yet release arm64 binaries (such as grpc 2.25.0).

Basically, I'm stuck in version hell, on the bleeding edge of users.

**Comment by danmosemsft** [comment]
Oh dear. cc @ooooolivia who did #1403

**Comment by adamsitnik** [comment]
@RobertHenry6bev which asp.net benchmarks are you trying to run?

**Comment by ooooolivia** [comment]
@DrewScoggins  @billwert

**Comment by adamsitnik** [comment]
@RobertHenry6bev are you trying to run the ASP.NET benchmarks from https://github.com/aspnet/Benchmarks/ repo? or you are trying to run the microbenchmarks from this repo using a BenchmarkClient from https://github.com/aspnet/Benchmark repo?

**Comment by sebastienros** [comment]
> I'm encountering situations where the various dotnet/aspnet benchmarks implicitly or explicitly refer to amd64(eg x86_64) binaries, rather than ARM64 binaries.

Can you provide an example? On another related issue I already mentioned that our clients don't work on ARM, you can only use then on Lin/Win, but it doesn't prevent you from benchmarking the app on the ARM machine, just the load generators. We run most of our benchmarks on ARM too.

> Or else reference packages on the web that don't yet release arm64 binaries (such as grpc 2.25.0).

For GRPC, we only test on Win/Lin. However the clients and servers have .NET implementation that should work fine. Have you tried these instead of the native ones? If you still want to run native grpc benchmarks on ARM I can point you to the dev who owns these benchmarks.

**Comment by RobertHenry6bev** [comment]
I am trying to run the aspnet benchmarks from https://github.com/aspnet/Benchmarks.git using the legacy docker client/agent/db/server setup.

Ach. I have stupidly regressed to running client on the arm.  I'll restructure my ensemble.

**Comment by adamsitnik** [comment]
> I am trying to run the aspnet benchmarks from https://github.com/aspnet/Benchmarks.git

this repo (`dotnet/performance`) and `aspnet/Benchmarks` are two independent repositories.

@RobertHenry6bev Could you please open an issue in https://github.com/aspnet/Benchmarks repo then?
